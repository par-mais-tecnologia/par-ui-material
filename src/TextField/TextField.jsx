import React, { PureComponent } from 'react'
import FormControl from '@material-ui/core/FormControl/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText/FormHelperText'
import TextFieldMUI from '@material-ui/core/TextField/TextField'
import PropTypes from 'prop-types'

class TextField extends PureComponent {
  state = {}

  validationInstance = this.props.validator ? this.props.validator.validator.register(
    this
  ) : false

  handleInputBlur (event) {
    if (this.props.onBlur) {
      return this.props.onBlur(event)
    } else {
      return this.validationInstance ? this.validationInstance.validate(event) : false
    }
  }

  handleInputChange (event) {
    if (this.props.onChange) {
      this.props.onChange(event)
    }
  }

  handleInputFocus (event) {
    if (this.props.onFocus) {
      this.props.onFocus(event)
    }
  }

  getStyle (props) {
    const { minWidth, maxWidth, style } = props
    return {
      minWidth: minWidth || '0',
      maxWidth: maxWidth || '100%',
      ...style
    }
  }

  render () {
    const { helperText, id, label, required, showHelper, value, type, autoFocus, spellCheck } = this.props

    const inputProps = {
      onBlur: this.handleInputBlur.bind(this),
      onChange: this.handleInputChange.bind(this),
      onFocus: this.handleInputFocus.bind(this),
      spellCheck: spellCheck
    }

    if (this.validationInstance) {
      this.state.errors = {
        ...this.validationInstance.error
      }
    } else {
      this.state.errors = { hasError: false }
    }

    const proxyProps = { ...this.props }
    delete proxyProps.validatorType
    delete proxyProps.showHelper
    delete proxyProps.minWidth
    delete proxyProps.maxWidth

    return (
      <FormControl
        aria-describedby={id}
        required={required}
        style={this.getStyle(this.props)}>
        <TextFieldMUI
          error={this.state.errors.hasError}
          {...proxyProps}
          {...inputProps}
          label={label}
          id={id}
          margin='normal'
          value={value}
          type={type}
          autoFocus={autoFocus}
        />
        {showHelper ? <FormHelperText id={id}> {helperText} </FormHelperText> : ''}
        {this.state.errors.hasError ? <FormHelperText
          id={id}> {this.state.errors.errorMessage} </FormHelperText> : ''}
      </FormControl>
    )
  }
}

TextField.propTypes = {
  spellCheck: PropTypes.bool
}

TextField.defaultProps = {
  id: `textField`,
  label: '',
  required: false,
  showHelper: false,
  value: '',
  margin: 'normal',
  spellCheck: false
}

export default TextField
