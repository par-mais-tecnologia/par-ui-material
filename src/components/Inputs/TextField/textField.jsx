import React, { PureComponent } from 'react'
import * as dictionary from '../../Core/dictionary'
import FormControl from '@material-ui/core/FormControl/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText/FormHelperText'
import TextFieldMUI from '@material-ui/core/TextField/TextField'
import * as Validation from '../../Core/validation'

class TextField extends PureComponent {
  state = {}

  handleInputBlur(event) {
      if (this.props.onBlur) {
          this.props.onBlur(event)
      }
  }

  handleInputChange(event) {
      if (this.props.onBlur) {
          this.props.onChange(event)
      }
  }

  handleInputFocus(event) {
      if (this.props.onBlur) {
          this.props.onFocus(event)
      }
  }

  render () {
      const inputProps = {
          onBlur: this.handleInputBlur.bind(this),
          onChange: this.handleInputChange.bind(this),
          onFocus: this.handleInputFocus.bind(this),
      };

    const { value, showHelper, required, id } = this.props

    return (
      <FormControl
        aria-describedby='email-form-control'
        required={required}>
        <TextFieldMUI
          {...inputProps}
          value={value}
          id={id}
          label={dictionary.EMAIL}
          margin='normal'
        />
        {showHelper ? <FormHelperText id='email-form-control'> {errorMessage} </FormHelperText> : ''}
      </FormControl>
    )
  }
}

TextField.defaultProps = {
  label: dictionary.EMAIL,
  value: '',
  showError: true,
  required: false,
  errorMessage: '',
  hasError: false
}

export default TextField
