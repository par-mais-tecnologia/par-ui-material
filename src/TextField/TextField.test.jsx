import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import TextField from './'

configure({ adapter: new Adapter() })

export const test = describe('TextField', () => {
  it('shallow TextField', () => {
    const wrapper = shallow(
      <TextField />
    )
    expect(wrapper).toBeTruthy()
  })

  it('shallow TextField match Snapshot', () => {
    const wrapper = shallow(
      <TextField />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
