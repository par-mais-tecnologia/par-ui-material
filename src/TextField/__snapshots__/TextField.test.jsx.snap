// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField shallow TextField match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    id="textField 24"
    label=""
    required={false}
    showHelper={false}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-describedby": "textField 24",
      "children": Array [
        <TextField
          error={false}
          id="textField 24"
          label=""
          margin="normal"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          select={false}
          showHelper={false}
          value=""
          variant="standard"
        />,
        "",
        "",
      ],
      "required": false,
      "style": Object {
        "maxWidth": "100%",
        "minWidth": "0",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "error": false,
          "id": "textField 24",
          "label": "",
          "margin": "normal",
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "required": false,
          "select": false,
          "showHelper": false,
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
      "",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-describedby": "textField 24",
        "children": Array [
          <TextField
            error={false}
            id="textField 24"
            label=""
            margin="normal"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            select={false}
            showHelper={false}
            value=""
            variant="standard"
          />,
          "",
          "",
        ],
        "required": false,
        "style": Object {
          "maxWidth": "100%",
          "minWidth": "0",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "error": false,
            "id": "textField 24",
            "label": "",
            "margin": "normal",
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "required": false,
            "select": false,
            "showHelper": false,
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        "",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
